.github/workflows/build-ui.yml

```diff
@@ -12,12 +12,17 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - name: Checkout
+      - name: Checkout current repo head to support automated local PRs
+        if: github.event.pull_request.head.repo.full_name == github.repository
         uses: actions/checkout@v4.1.1
         with:
           ref: ${{ github.head_ref }}
           fetch-depth: 2
 
+      - name: Checkout default if not current repo
```

Comment:
You mean to say upstream repo instead of current?

Reply:
That makes sense. I'll make that change